### PR TITLE
make entities easier to stack

### PR DESCRIPTION
**Pull request by chmodsayshello**
This small pull request makes it easier to stack a lot a of players on top of each other (for example for a fun screenshot)...
Rather than ejecting entity XY when a player clicks the entity it's sitting on, it now checks in a loop for the next entity in the stack that has no passenger, and mounts the player onto that...

### CODE REVIEW

**Comment by chmodsayshello** [comment]
Forgot to add a commit

**Comment by moderatorman** [comment]
so i tested your changes... i phased out of reality before timing out of the test server...

**Comment by chmodsayshello** [comment]
Even after redoing all my tests, i couldn't reproduce said behaviour, tested on localhost with 5 players :/

What does the log say?

> so i tested your changes... i phased out of reality before timing out of the test server...

**Comment by moderatorman** [comment]
all i did was mount then dismount a sheep lol
edit: no logs/errors

**Comment by chmodsayshello** [comment]
Tried a chicken rn, still notthing, let me test  a sheep, in case its a special case

**Comment by moderatorman** [comment]
here's a video

https://user-images.githubusercontent.com/12193288/216157484-69b534ea-e5cf-429b-ac78-aaee8c1a3e4a.mp4

**Comment by chmodsayshello** [comment]
OOH, I think i found it, this also gets called when the player ejects from the entity

File: src/main/java/net/oldschoolminecraft/pm/PlayerHandler.java [hunk 1]
```diff
@@ -35,7 +35,11 @@ public void onPlayerInteractEntity(final PlayerInteractEntityEvent event) {
                 return;
             }
             if (ent.getPassenger() != null) {
-                ent.eject();
```
Comment (chmodsayshello) [inline src/main/java/net/oldschoolminecraft/pm/PlayerHandler.java:38]:
```
if(ent.getPassenger().equals(event.getPlayer()){
     ent.eject();
     return;
}
```
(called before the loop, to check if its a player trying to mount or dismount)

**Comment by chmodsayshello** [comment]
I came up with a possible solution, will test it tomorrow or so

**Comment by moderatorman** [comment]
i tested it for you, and it works!

**Pull request merged**